torch/distributed/distributed_c10d.py

```diff
@@ -1329,6 +1335,21 @@ def _get_split_source(pg):
 
     return split_from
 
+def _shutdown_backend(pg):
+    """
+    Try to shut down the backend of a process group.
+    Currently, only ProcessGroupNCCL backend is supported.
+    No op for other backends.
+    """
+    backend = None
+    try:
+        backend = pg._get_backend(torch.device("cuda"))
+    except RuntimeError:
+        pass
+    if is_nccl_available() and isinstance(backend, ProcessGroupNCCL):
```

Comment:
This feels a bit heuristic/arbitrary.
If an `if-else` is truly needed, shall we put it upfront as much as possible than wrapping it in some function whose name sounds generic?

Reply:
I think is_nccl_available() is not needed here as we are checking isinstance(backend, ProcessGroupNCCL) anyway. "cuda" is needed to check the backend used for GPU devices.